pipelines/controller/src/test/java/org/openmrs/analytics/DataPropertiesTest.java

```diff
@@ -15,18 +15,34 @@
  */
 package org.openmrs.analytics;
 
+import java.io.IOException;
+import okhttp3.mockwebserver.MockWebServer;
 import org.junit.Assert;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.Test;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.test.autoconfigure.actuate.observability.AutoConfigureObservability;
 import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.TestPropertySource;
 
 @SpringBootTest
 @AutoConfigureObservability
+@TestPropertySource("classpath:application-test.properties")
 public class DataPropertiesTest {
 
   @Autowired private DataProperties dataProperties;
 
+  private static MockWebServer mockFhirServer;
+
+  @BeforeAll
+  public static void setUp() throws IOException {
+    mockFhirServer = new MockWebServer();
+    mockFhirServer.start(9091);
+    mockFhirServer.enqueue(MockUtil.getMockResponse("data/fhir-metadata-sample.json"));
+    mockFhirServer.enqueue(MockUtil.getMockResponse("data/patient-count-sample.json"));
```

Comment:
Question: Are these responses returned in sequence? I mean regardless of what the request is, these are returned in sequence? If yes, is there a way to make this request dependent. I am just wondering as I don't know `MockWebServer`, we don't need to change it in this PR.

Reply:
Yes, by default the responses are returned in sequence. We can customise the behaviour to return responses based on request path using this [dispatcher](https://github.com/square/okhttp/tree/master/mockwebserver#dispatcher), but this looks little non trivial.

Another trivial which can be used as a mock server is this [Wiremock](https://www.baeldung.com/introduction-to-wiremock) which provides APIs to return responses based on request.

Reply:
Closing this comment for now, will evaluate more and make changes if required in a new PR.